### PR TITLE
Jetty 9.4.x 1803 proposal 0+2

### PR DESCRIPTION
**Pull request by gregw**
For #1803 
This version of EWYK has been JMH micro benchmarked to be at least as good as PEC in multiple modes, including non blocking invocation so that it can be used with HTTP/2 without thread starvation.

While further benchmarking is required to prove the macro benefit of EWYK, I think it worthwhile to consider merging this now as it removes the exceptional handling for HTTP/2 and builds on the good simplification of reserved threads already merged in proposal 0.

### CODE REVIEW

File: jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/AbstractHTTP2ServerConnectionFactory.java [hunk 1]
```diff
@@ -142,7 +142,7 @@ public void setStreamIdleTimeout(long streamIdleTimeout)
 
     /**
      * @return -1
-     * @deprecated
+     * @deprecated not used anymore
```
Comment (joakime) [inline jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/AbstractHTTP2ServerConnectionFactory.java]: Typically, good `@deprecated` javadoc will include a statement indicating replacement. (even if it has "no replacement")

See: http://www.oracle.com/technetwork/java/javase/tech/index-137868.html#@deprecated

Comment (sbordet) [reply jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/AbstractHTTP2ServerConnectionFactory.java]: What there is ambiguous in "not used anymore" ?

Comment (joakime) [reply jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/AbstractHTTP2ServerConnectionFactory.java]: think about the consumers of this information, those updating their usages of this class/method.
that just means it's not used here, in this class. 
was it moved somewhere else? 
has this functionality been moved (is available from) somewhere else?
is there an alternative technique?

Comment (sbordet) [reply jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/AbstractHTTP2ServerConnectionFactory.java]: Wording suggestions ? "feature removed, don't use it, no replacement" ?

Comment (joakime) [reply jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/AbstractHTTP2ServerConnectionFactory.java]: +1 for that wording. its clear to me.
@WalkerWatch your thoughts?

Comment (WalkerWatch) [reply jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/AbstractHTTP2ServerConnectionFactory.java]: As Oracle suggests we should probably add, going forward, when it was deprecated. As for the wording in this instance, I like the suggestion from @sbordet. "@deprecated As of Jetty X.x.x, feature removed, no replacement."

**Pull request merged**